### PR TITLE
Offline access

### PR DESCRIPTION
**Pull request by WcaleNieWolny**

### CODE REVIEW

**Comment by WcaleNieWolny** [comment]
Do not merge it just yet, offline access is a WiP

**Comment by riderx** [comment]
We need to support same min version as capacitor

File: src/definitions.ts [hunk 1]
```diff
@@ -87,17 +87,22 @@ export interface GoogleLoginOptions {
   grantOfflineAccess?: boolean;
 }
 
-export interface GoogleLoginResponse {
+export interface GoogleLoginOnlineResponse {
   accessToken: AccessToken | null;
-  idToken: string | null;
+  responseType: 'online';
```
Comment (riderx) [inline src/definitions.ts:92]: why idToekn is removed? It's use currently offline shouldn't disturb current usage

File: android/src/main/java/ee/forgr/capacitor/social/login/SocialLoginPlugin.java [hunk 2]
```diff
@@ -54,6 +56,11 @@ public void initialize(PluginCall call) {
 
     JSObject google = call.getObject("google");
     if (google != null) {
+      if (android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.N) {
```
Comment (riderx) [inline android/src/main/java/ee/forgr/capacitor/social/login/SocialLoginPlugin.java:59]: this should be remove and use alternative code to make it work same version as capacitor

File: docs/setup_google.md [hunk 3]
```diff
@@ -264,49 +294,154 @@ In this part, you will learn how to setup Google login in Android
...
+        if (pluginHandle == null) {
+            Log.i("Google Activity Result", "SocialLogin login handle is null");
+            return;
+        }
+        Plugin plugin = pluginHandle.getInstance();
+        if (!(plugin instanceof SocialLoginPlugin)) {
+            Log.i("Google Activity Result", "SocialLogin plugin instance is not SocialLoginPlugin");
+            return;
+        }
+        ((SocialLoginPlugin) plugin).handleGoogleLoginIntent(requestCode, data);
```
Comment (riderx) [inline docs/setup_google.md:328]: can you explain why this is required? in what scenario?

Comment (WcaleNieWolny) [reply docs/setup_google.md:328]: Both offline access and additional scopes are based on authorization. Authorization requires this change

File: android/src/main/java/ee/forgr/capacitor/social/login/SocialLoginPlugin.java (hunk 2 above)
Comment (WcaleNieWolny) [reply android/src/main/java/ee/forgr/capacitor/social/login/SocialLoginPlugin.java:59]: I never pulled latest "development" branch. There, the problem does not happen

**Comment by WcaleNieWolny** [comment]
Closing in favor of #56

**Pull request closed without merging**